workflow/train_test_split.py

```diff
@@ -0,0 +1,167 @@
+from collections.abc import Iterable
+import numpy as np
+import pandas as pd
+import pytest
+
+
+def train_test_split(
+    df,
+    unique_key,
+    test_size=0.2,
+    seed=0,
+    stratify=None
+):
+    if stratify is None:
+        test_df = get_test_split(df, unique_key, test_size, seed)
+    else:
+        test_df = stratified_split(df, unique_key, stratify, test_size, seed)
+
+    train_df = df[~df[unique_key].isin(test_df[unique_key])]
+
+    return train_df, test_df
+
+
+def get_test_split(df, unique_key, test_size, seed=0, resolution=100000):
+    split_unique_key = df[unique_key].apply(
+        lambda x: ((hash(hash(x) + seed) % resolution))
+    )
+    return df[split_unique_key < resolution * test_size]
+
+
+def stratified_split(df, unique_key, stratify, test_size, seed):
+
+    test_dfs = []
+    for label in get_labels(df, stratify):
+        if type(stratify) is str:
+            df_part = df[df[stratify] == label]
```

Comment:
```suggestion
            df_accepted = df[df[stratify] == label]
```
or something a bit more descriptive to make it easier to follow

Reply:
`accepted` borrowed from https://en.wikipedia.org/wiki/Rejection_sampling which is perhaps not the best